pipeline/src/pipeline.rs

```diff
@@ -953,81 +904,75 @@ impl<T: FieldElement> Pipeline<T> {
         &mut self,
         mut writer: W,
     ) -> Result<(), Vec<String>> {
-        self.advance_to(Stage::PilWithEvaluatedFixedCols)?;
-        match self.artifact.as_ref().unwrap() {
-            Artifact::PilWithEvaluatedFixedCols(PilWithEvaluatedFixedCols { pil, fixed_cols }) => {
-                let backend = self
-                    .arguments
-                    .backend
-                    .expect("backend must be set before generating verification key!");
-                let factory = backend.factory::<T>();
-
-                let mut setup_file = self
-                    .arguments
-                    .setup_file
-                    .as_ref()
-                    .map(|path| BufReader::new(fs::File::open(path).unwrap()));
-
-                let backend = factory
-                    .create(
-                        pil.borrow(),
-                        &fixed_cols[..],
-                        self.output_dir(),
-                        setup_file
-                            .as_mut()
-                            .map(|file| file as &mut dyn std::io::Read),
-                        None,
-                    )
-                    .unwrap();
-
-                match backend.export_verification_key(&mut writer) {
-                    Ok(()) => Ok(()),
-                    Err(powdr_backend::Error::BackendError(e)) => Err(vec![e]),
-                    _ => panic!(),
-                }
-            }
+        let backend = self
```

Comment:
I actually started that in this PR, but I stopped because I'm not sure anymore Pipeline should hold the verification key since for the eSTARK case it is quite big. But yea we can think about how to do it in a different PR. This PR still gives the advantage of accessing the artifacts at the same time without having to clone Pipeline and start from scratch which is nice for applications (like powdr-revme)

Reply:
OK, do we understand why it's big? If the proof is succinct, it should be nowhere close to the witness size, which we already have in the pipeline.